draft-ietf-core-problem-details.md

```diff
@@ -13,224 +17,408 @@ stand_alone: yes
 
 # Basic Problem Details
 
-A CoRE Problem Details is a CoRAL document with the following elements:
+A Concise Problem Details data item is a CBOR data item with the following
+structure (notated in CDDL {{!RFC8610}}):
+
+~~~ CDDL
+problem-details = non-empty<{
+  ? &(title: -1) => text
+  ? &(detail: -2) => text
+  ? &(instance: -3) => ~uri
+  ? &(response-code: -4) => uint .size 1
+  standard-problem-detail-entries
+  custom-problem-detail-entries
+}>
+
+standard-problem-detail-entries = (
+  * nint => any
+)
+
+custom-problem-detail-entries = (
+  * (uint/~uri) => { + any => any }
+)
+
+non-empty<M> = (M) .and ({ + any => any })
+~~~
+{: #cddl title="Problem Detail Data Item"}
+
+A number of problem detail entries, the Standard Problem Detail
+entries, are predefined (more predefined details can be registered,
+see {{new-spdk}}):
 
-* "type" (id) - The problem type.  This is a mandatory element.
-* "title" (text) - A short, human-readable summary of the problem type.  It SHOULD NOT change from occurrence to occurrence of the problem.
-* "detail" (text) - A human-readable explanation specific to this occurrence of the problem.
-* "instance" (uri) - A URI reference that identifies the specific occurrence of the problem.  It may or may not yield further information if dereferenced.
+{:vspace}
+The title (key -1):
+: A short, human-readable summary of the problem type.
+  It SHOULD NOT change from occurrence to occurrence of the problem.
 
-Consumers MUST use "type" as primary identifiers for the problem type; the "title" string is advisory and included only for consumers who are not aware of the semantics of the "type" value.  An initial set of "type"s and corresponding "title"s is defined in {{sec-response-code-types}}.
+The detail (key -2):
+: A human-readable explanation specific to this occurrence of the problem.
+
+The instance (key -3):
+: A URI reference that identifies the specific occurrence of the problem.
+  It may or may not yield further information if dereferenced.
+
+The response-code (key -4)
+: The CoAP response code ({{Section 5.9 of -coap}}) generated by the origin
+  server for this occurrence of the problem.
+
+Note that unlike {{RFC7807}} Concise Problem Details data items have no
+explicit type.
+
+The "title" string is advisory and included to give
+consumers a shorthand for the category of the error encountered.
 
 The "detail" member, if present, ought to focus on helping the client correct the problem, rather than giving debugging information.  Consumers SHOULD NOT parse the "detail" member for information; extensions (see {{sec-new-attributes}}) are more suitable and less error-prone ways to obtain such information.
 
-Note that the "instance" URI reference may be relative; this means that it must be resolved relative to the document's base URI, as per {{!I-D.ietf-core-coral}}.
+Note that the "instance" URI reference may be relative; this means
+that it must be resolved relative to the document's base URI, as per
```

Comment:
OK, we probably need to decide whether this thing is a document, or a data item, or a representation, or...